### ISSUE TITLE
Pen-only device does not work correctly in Chrome - ignores the touch events

### ISSUE THREAD

**Issue by IanYates** [fabricjs/fabric.js#4473]
(this doesn't quite fit into the bug template so I've just told the story and offered up three ideas for a fix)

Using the Kitchen Sink demo on the website, running latest 2.0rc1 code (as at 2017-11-20).  In our app, where we're experimenting with fabric v1.7, we've observed the same behaviour.  All code below is as per the demo on the website using v2.

We have Windows 10, 1707, running Chrome latest stable release - 62.0.3202.94 (Official Build) (64-bit) (cohort: Stable)

Our desktop PC has a drawing tablet attached as a third monitor - note that it is a *pen only* device, no touch (and the other two desktop screens are not touch either)

In fabric.js of the demo site, line 30 has this
```
fabric.isTouchSupported = 'ontouchstart' in fabric.window;
```

This ends up being false.

However, when we then use the pen on the canvas, touch events are actually made.

The problem is that we get paths made, but those paths are empty.

That's because at line 2187 we have this
```
  if (fabric.isTouchSupported) {
    pointerX = function(event) {
      return _getPointer(event, 'pageX', 'clientX');
    };
    pointerY = function(event) {
      return _getPointer(event, 'pageY', 'clientY');
    };
  }
```

Since `isTouchSupported` is false, the patching of `pointerX` and `pointerY` functions is not done to try to use touch events first.
Then, because of that, the original functions (defined just above at line 2170), are still used.  They don't work on a touch event, resulting in `{x: NaN, y: NaN}` being the result of calls to getPointer.

We've just forced isTouchSupported to true and all works well.  From what I can see, the patched versions try to use any available touch properties on the events and then gracefully fall back to using clientX and clientY as you'd get on a regular mouse event anyway.

So...  What's the best way for us to patch fabric with this support?  I can think of a couple of options

1. Always use a form of the pointerX and pointerY functions that try to inspect the events as touch events first.  I appreciate this may slightly slow down performance (or at least that's the only reason I can think of for this optional patching of the functions in the first place)
2. Before initialising fabric.js on our page, make a dummy call to window.ontouchstart = undefined.  This will fool fabric.js' initialisation code to think touch events are present.  This feels like a gross hack though.
3. Update fabric.js' initialisation code so that it can check for some global variable like `__fabricAssumeTouch__` in addition to looking for the browser window's `ontouchstart` function to be present.

Which makes the most sense to you?

Three other things of note which, I think, support my conclusions above.

1. In Edge things work.  Fabric still thinks that touch support isn't present.  However the type of event coming through to `getPointer()` is a mouse event rather than a touch event so things still work as expected.

2. If we go to chrome://flags and use the option to force the touch events API to always be accessible, then fabric assumes it's in a touch environment and things work as expected

3. On a Microsoft Surface, which has pen and touch, Chrome, without needing to set any flags, has the properties on the window object that fabric detects, so pen input works on a Surface in Chrome.

Thank you!!!

**Comment by asturur** [comment]
So if i understood good is that the touch device is not recognized as a pointer event emitter from the browser.

Now you are developing an app, and possibly a user has something that works as your pen input device but does not trigger isTouchSupported to true.

I think that the safest way is to give devs a way to enable it and pay for the perfomance drop ( even if is small ) if they want.

This can be done in a hacky way at init time like
`fabric.isTouchSupported = '_forceFabricPointerEvent' in fabric.window || ...`

but is not like super nice, or refactor the functions in dom_events.js so that we get a

fabric.enablePointerEvents()  and fabric.disablePointerEvents() that can move back and forward the 2 implementations.

i'm open to ideas that are clean and performant.

**Comment by IanYates** [comment]
I'll experiment a bit more and see what we come up with.  Thanks!

I saw in a separate issue that you're suggested people start with v2.x.  I think we may look to move to it too then since we've not yet released anything for end-users that's using fabric.  Whatever experimentation we do I'll do against 2.x and report back.

**Comment by asturur** [comment]
so as of now in fabric 2.0 there may be bugs for recent changes but you find bugs with the same probability in the stable 1.7 branch.

i ll probably rush to fix open bugs and release 2.0.0
